engines/scumm/imuse_digi/dimuse_engine.cpp

```diff
@@ -0,0 +1,760 @@
+}
+
+void IMuseDigital::diMUSESaveLoad(Common::Serializer &ser) {
+	cmdsSaveLoad(ser);
+	return;
+}
+
+int IMuseDigital::diMUSESetGroupVol(int groupId, int volume) {
+	return cmdsHandleCmd(7, groupId, volume, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEStartSound(int soundId, int priority) {
+	return cmdsHandleCmd(8, soundId, priority, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEStopSound(int soundId) {
+	debug(5, "IMuseDigital::diMUSEStopSound(): %d", soundId);
+	return cmdsHandleCmd(9, soundId, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEStopAllSounds() {
+	debug(5, "IMuseDigital::diMUSEStopAllSounds()");
+	return cmdsHandleCmd(10, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEGetNextSound(int soundId) {
+	return cmdsHandleCmd(11, soundId, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSESetParam(int soundId, int paramId, int value) {
+	return cmdsHandleCmd(12, soundId, paramId, value, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEGetParam(int soundId, int paramId) {
+	return cmdsHandleCmd(13, soundId, paramId, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEFadeParam(int soundId, int opcode, int destValue, int fadeLength) {
+	return cmdsHandleCmd(14, soundId, opcode, destValue, fadeLength, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSESetHook(int soundId, int hookId) {
+	return cmdsHandleCmd(15, soundId, hookId, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSESetTrigger(int soundId, int marker, int opcode, int d, int e, int f, int g, int h, int i, int j, int k, int l, int m, int n) {
+	return cmdsHandleCmd(17, soundId, marker, opcode, d, e, f, g, h, i, j, k, l, m, n);
+}
+
+int IMuseDigital::diMUSEStartStream(int soundId, int priority, int bufferId) {
+ 	return cmdsHandleCmd(25, soundId, priority, bufferId, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSESwitchStream(int oldSoundId, int newSoundId, int fadeDelay, int fadeSyncFlag2, int fadeSyncFlag1) {
+	return cmdsHandleCmd(26, oldSoundId, newSoundId, fadeDelay, fadeSyncFlag2, fadeSyncFlag1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+// Variation for FT and DIG demo
+int IMuseDigital::diMUSESwitchStream(int oldSoundId, int newSoundId, uint8 *crossfadeBuffer, int crossfadeBufferSize, int vocLoopFlag) {
+	return cmdsHandleCmd(26, oldSoundId, newSoundId, (uintptr)crossfadeBuffer, crossfadeBufferSize, vocLoopFlag, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEProcessStreams() {
+	return cmdsHandleCmd(27, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1);
+}
+
+int IMuseDigital::diMUSEQueryStream(int soundId, int &bufSize, int &criticalSize, int &freeSpace, int &paused) {
```

Comment:
I don't see usage diMUSEQueryStream()

Reply:
Whoops, I forgot to remove that, its functionality only used internally with streamerQueryStream, within the Dispatch module